packages/remix-react/transition.ts

```diff
@@ -630,17 +630,22 @@ export function createTransitionManager(init: TransitionManagerInit) {
     return false;
   }
 
-  function getFetcherRequestMatch(
-    url: URL,
-    matches: RouteMatch<ClientRoute>[]
-  ) {
+  function getRequestMatch(url: URL, matches: ClientMatch[]) {
     let match = matches.slice(-1)[0];
 
-    if (!isIndexRequestUrl(url) && match.route.index) {
-      return matches.slice(-2)[0];
+    if (isIndexRequestUrl(url) && match.route.index) {
+      return match;
```

Comment:
Return the leaf for index requests, otherwise return the lowest path contributing match